config.dev.toml

```diff
@@ -166,6 +166,35 @@ BasicAuthUser = ""
 # Env override: BASIC_AUTH_PASS
 BasicAuthPass = ""
 
+# PropagateAuth, when set to true, will pass the Basic Authentication
+# Headers to the "go mod download" operations. This will allow a user
+# to pass their credentials for a private repository and have Athens be
+# able to download and store it. Note that, once a private repository is stored,
+# Athens will naively serve it to anyone who requests it.
+# 
+# Therefore, it is **important** that you
+# make sure you have a ValidatorHook or put Athens behind an auth proxy that always 
+# ensures access to modules are securely authorized.
+# 
+# Note that "go mod download" uses "git clone" which will look for these credentials 
+# in the $HOME directory of the process. Therefore, turning this feature on means that each
+# "go mod download" will have its own $HOME direcotry with only the .netrc file. If
+# your "go mod download" relies on your global $HOME directory (such as .gitconfig), then
+# you must turn this feature off. If you'd like to specify files to be copied from the global
+# $HOME directory to the temporary one, please open an issue at https://github.com/gomods/athens
+# to gauge demand for such a feature before implementing.
+# 
+# Env override: ATHENS_PROPAGATE_AUTH
+PropagateAuth = false
+
+# PropagateAuthPatterns can be used to limit which modules will have
+# the credentials accompanied when calling "go mod download". Note that
+# this is only when a module is not in storage. See PropagateAuth above
+# for more details.
+# 
+# Env override: ATHENS_PROPAGATE_AUTH_PATTERNS
+PropagateAuthPatterns = ["*"]
```

Comment:
I'm not sure if I like the default here, from a security perspective. Perhaps just `["localhost"]`?

Reply:
@twexler you're right. I removed this whole functionality in favor of static hosts. This should eliminate the possibility of accidentally leaking credentials to any host other host than the pre declared one.